host.go

```diff
@@ -109,6 +109,24 @@ func (p *Plugin) currentPlugin() *CurrentPlugin {
 	return &CurrentPlugin{p}
 }
 
+// SetHostFunctionError allows the host function to set an error that will be
+// gracefully returned by extism guest modules.
+func (p *CurrentPlugin) SetHostFunctionError(ctx context.Context, err error) {
```

Comment:
Since they're both using the same underlying extism function, perhaps we can match go pdk's naming by calling this `SetError`